Add recipe for wavedrom-mode

gmlarumbe opened the pull request:
### Brief summary of what the package does

Major-mode for editing and rendering WaveJSON files to create timing diagrams using [Wavedrom](https://github.com/wavedrom/wavedrom).

### Direct link to the package repository

https://github.com/gmlarumbe/wavedrom-mode

### Your association with the package

Author/maintainer.

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [X] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [X] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [X] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [X] My elisp byte-compiles cleanly
- [X] I've used `M-x checkdoc` to check the package's documentation strings
- [X] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

riscy commented:
Thanks for this.  On a quick first pass --

`wavedrom-mode.el` with byte-compile using Emacs 28.1:
```
wavedrom-mode.el:214:36: Warning: the function `js-json-mode' is not known to
    be defined.
```
This might be due to a missing `Package-Requires` or `require`?

gmlarumbe commented:
Hi @riscy ,

Thanks for your review.

`js-json-mode` is provided by `js` in Emacs 29.1. I updated the `Package-Requires` header and added this line at the beginning:
```elisp
(require 'js)
```

I tested those changes with melpazoid GitHub Action but I still get the same warning since `apt-get` seems to install Emacs 28.1 in `ubuntu-latest`. 

Should I change or add anything?

Thanks!

riscy commented:
I suspected this might be the case, but now that we've got emacs 29.1 as a dependency this is fine - merging!

(And yes, melpazoid is still on Emacs 28.1 for now.)
